framework/keyring-interface.md

```diff
@@ -36,37 +36,48 @@ in this document are to be interpreted as described in [RFC 2119](https://tools.
 
 ### OnEncrypt
 
-This interface takes [encryption materials](#structures.md#encryption-materials) as input
-and MAY modify it with any of the following behaviors:
+The following inputs are REQUIRED:
+
+- [Algorithm Suite](#algorithm-suites.md)
+- [Encryption Context](#encryption-context)
+
+The following inputs are OPTIONAL:
+
+- [Plaintext Data Key](#structures.md#plaintext-data-key)
+
+This interface MAY return [data key materials](#structures.md#data-key-materials) appropriate for the request. If it does return this output
+and a plaintext data key was provided as input, the output must specify the same plaintext data key.
```

Comment:
And in that case I'd rather not return it at all from Keyring operations since that's simpler. :) I'm starting to lean more towards just returning multiple values rather than a structure.

Reply:
That's where I'm leaning too, and in that case we'd need to ensure that we specify that the outputted plaintextDataKey meets the spec for the AlgorithmSuiteID on input, since it would no longer be tied to the plaintextDataKey through the DataKeyMaterials.